### DISCUSSION TITLE
Example of session while using tower_cookies?

### DISCUSSION THREAD

**Discussion by SylvainMartel** [tokio-rs/axum#695] [Q&A] [answered]
Hi, when using the `flash` crate, we also have to use the crate `tower_cookies` too for cookie management  As such, the session example in the repo will not work(I remember reading somewhere that tower_cookies kinda override any cookies made by other means?).   

Right now, I have no idea how to get current cookies while inside the FromRequest implementation.  When I do a `cookies.list())`  in the FromRequest implementation, it shows an Empty list even though there are 2 cookies that were created upon the user logging in(and we can see them in the browser.)  

So it would be nice to have another example using tower_cookies for reference :)

**Comment by davidpdrsn** [comment, marked as answer]
Wanna re-ask this question for tower-cookies? I’m not familiar enough with its internals to know.

**Comment by SylvainMartel** [comment]
In case someone else is looking for it, the example on how to get access to the tower-cookies middleware inside the FromRequest implementation is [here](https://github.com/imbolc/tower-cookies/blob/main/examples/counter-extractor.rs)  Thanks to @imbolc for his time on this, I learned alot from it 👍
